test/lib.bash

```diff
@@ -115,12 +115,131 @@ function downstream_eventing_e2e_tests {
   return $failed
 }
 
+# == Upgrade testing
+
+function run_rolling_upgrade_tests {
+  logger.info "Running rolling upgrade tests"
+  (
+  local latest_cluster_version latest_serving_version latest_eventing_version \
+    rootdir scope serving_in_scope eventing_in_scope serving_prober_pid \
+    eventing_prober_pid prev_serving_version prev_eventing_version \
+    ocp_target_version
+
+  scope="${1:?Provide an upgrade scope as arg[1]}"
+  serving_in_scope="$(echo "${scope}" | grep -vq serving ; echo "$?")"
+  eventing_in_scope="$(echo "${scope}" | grep -vq eventing ; echo "$?")"
+
+  prev_serving_version="$(actual_serving_version)"
+  prev_eventing_version="$(actual_eventing_version)"
+
+  # Save the rootdir before changing dir
+  rootdir="$(dirname "$(dirname "$(realpath "${BASH_SOURCE[0]}")")")"
+
+  (( serving_in_scope )) && prepare_knative_serving_tests
```

Comment:
Those instruction looks nice and clean, but they are dangerous as they hide errors.
```bash
(( serving_in_scope )) && something
```

Reply:
See: https://superuser.com/questions/1169664/bash-if-on-single-line?newreg=48d7979019cd4e1a87d8e7a8b461e7b6#comment1933718_1169669